plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/m2e/core/ui/wizard/MavenImportWizard.java

```diff
@@ -0,0 +1,148 @@
+package org.jboss.reddeer.eclipse.m2e.core.ui.wizard;
+
+import org.jboss.reddeer.common.condition.AbstractWaitCondition;
+import org.jboss.reddeer.common.wait.AbstractWait;
+import org.jboss.reddeer.common.wait.TimePeriod;
+import org.jboss.reddeer.common.wait.WaitUntil;
+import org.jboss.reddeer.common.wait.WaitWhile;
+import org.jboss.reddeer.core.condition.JobIsRunning;
+import org.jboss.reddeer.core.condition.ShellWithTextIsActive;
+import org.jboss.reddeer.core.condition.ShellWithTextIsAvailable;
+import org.jboss.reddeer.core.exception.CoreLayerException;
+import org.jboss.reddeer.jface.wizard.ImportWizardDialog;
+import org.jboss.reddeer.swt.exception.SWTLayerException;
+import org.jboss.reddeer.swt.impl.button.PushButton;
+import org.jboss.reddeer.swt.impl.combo.LabeledCombo;
+import org.jboss.reddeer.swt.impl.shell.DefaultShell;
+
+/**
+ * Wizard for importing an existing maven project.
+ * 
+ * @author apodhrad
+ *
+ */
+public class MavenImportWizard extends ImportWizardDialog {
+
+	public static final String TITLE = "Import Maven Projects";
+
+	/**
+	 * Constructs ImportWizard with "Maven > Existing Maven Projects".
+	 */
+	public MavenImportWizard() {
+		super("Maven", "Existing Maven Projects");
+	}
+
+	/**
+	 * Activate the wizard.
+	 */
+	public void activate() {
+		new DefaultShell(TITLE);
+	}
+
+	/**
+	 * Sets a root directory.
+	 * 
+	 * @param path
+	 *            Path
+	 */
+	public void setRootDirectory(String path) {
+		activate();
+		new LabeledCombo("Root Directory:").setText(path);
+	}
+
+	/**
+	 * Click on 'Refresh' button.
+	 */
+	public void refresh() {
```

Comment:
IMHO it is completely useless to divide the logic into Wizard and Page classes...